src/extensionVariables.ts

```diff
@@ -21,4 +22,10 @@ export namespace ext {
     export let tagFS: TagFileSystem;
     export let diagnosticWatcher: Disposable | undefined;
     export let diagnosticCollection: DiagnosticCollection;
+
+    export const resourceProviders: { [key: string]: ApplicationResourceProvider } = {};
+    export const groupByKeys: { [key: string]: string } = {
```

Comment:
Seems like these should maybe go into a constants file?

Reply:
Well these aren't "traditional" constants since the contents of the objects will be changing quite a bit, so I think they fit in more here.